homeassistant/components/ozw/light.py

```diff
@@ -140,29 +140,41 @@ def color_temp(self):
     def async_set_duration(self, **kwargs):
         """Set the transition time for the brightness value.
 
-        Zwave Dimming Duration values:
-        0       = instant
-        0-127   = 1 second to 127 seconds
-        128-254 = 1 minute to 127 minutes
-        255     = factory default
+        Zwave Dimming Duration values now use seconds as an
+        integer (max: 7620 seconds or 127 mins)
         """
         if self.values.dimming_duration is None:
             return
 
+        ozw_version = self.values.primary.ozw_instance.get_status().openzwave_version.split(
+            "."
+        )
+        ozw_build = int(ozw_version[2])
+        ozw_major_minor = float(".".join(ozw_version[0:2]))
+
         if ATTR_TRANSITION not in kwargs:
             # no transition specified by user, use defaults
-            new_value = 255
+            if ozw_build >= 1205 and ozw_major_minor >= 1.6:
```

Comment:
I think the check should be for the older version.

We should start by checking the major.minor version which should be 1.6 and then check the build version, which should be less than 1205. If the major.minor version is greater than 1.6 the build version doesn't matter.

Reply:
How about something like:
```
        if ATTR_TRANSITION not in kwargs:
            # no transition specified by user, use defaults
            new_value = 7621  # anything over 7620 uses the factory default
            if ozw_major_minor == 1.6 and ozw_build < 1205:
                new_value = 255  # default for older version
```

Reply:
Is this too simple?

```python
ozw_version = self.values.primary.ozw_instance.get_status().openzwave_version.split(".")[:3]

if ozw_version < [1, 6, 1206]:
    new_value = 255  # default for older version
```

Reply:
Still pretty simple. Sequence comparisons should do the logic you want.

```python
ozw_version = tuple(int(x) for x in self.values.primary.ozw_instance.get_status().openzwave_version.split("."))

if ozw_version < (1, 6, 1206):
```
Anyways, enough from me. :satisfied:

Reply:
I kinda like that solution, still have the old programming ways stuck in my head

Reply:
logic adjusted in latest commit